dotnet-authserver/src/TeacherIdentity.AuthServer/Jobs/PurgeConfirmationPinsJob.cs

```diff
@@ -0,0 +1,47 @@
+using TeacherIdentity.AuthServer.Models;
+
+namespace TeacherIdentity.AuthServer.Jobs;
+
+public class PurgeConfirmationPinsJob
+{
+    private static readonly TimeSpan _expiredPinGracePeriod = TimeSpan.FromDays(7);
+
+    private readonly TeacherIdentityServerDbContext _dbContext;
+
+    public PurgeConfirmationPinsJob(TeacherIdentityServerDbContext dbContext)
+    {
+        _dbContext = dbContext;
+    }
+
+    public async Task Execute(CancellationToken cancellationToken)
+    {
+        await DeleteEmailConfirmationPins(cancellationToken);
+        await DeleteSmsConfirmationPins(cancellationToken);
+    }
+
+    private async Task DeleteEmailConfirmationPins(CancellationToken cancellationToken)
+    {
+        var expiredEmailConfirmationPins = _dbContext.EmailConfirmationPins
+            .Where(p => p.Expires < DateTime.UtcNow - _expiredPinGracePeriod)
+            .ToList();
+
+        if (expiredEmailConfirmationPins.Any())
+        {
+            _dbContext.EmailConfirmationPins.RemoveRange(expiredEmailConfirmationPins);
+            await _dbContext.SaveChangesAsync(cancellationToken);
+        }
+    }
+
+    private async Task DeleteSmsConfirmationPins(CancellationToken cancellationToken)
+    {
+        var expiredSmsConfirmationPins = _dbContext.SmsConfirmationPins
+            .Where(p => p.Expires < DateTime.UtcNow - _expiredPinGracePeriod)
+            .ToList();
+
+        if (expiredSmsConfirmationPins.Any())
+        {
+            _dbContext.SmsConfirmationPins.RemoveRange(expiredSmsConfirmationPins);
+            await _dbContext.SaveChangesAsync(cancellationToken);
```

Comment:
I had some invalid mobile numbers in my DB from before the international prefix was added which caused this transaction to roll back. Do we want to make this into multiple transactions or are we happy with this?

Reply:
Could we do something like https://github.com/DFE-Digital/get-an-identity/blob/2891711a053b1cc7db9b055e4a5922e2404bee1a/dotnet-authserver/src/TeacherIdentity.AuthServer/Services/UserVerification/UserVerificationService.cs#L80 to avoid even pulling the records back at all?